src/services/swap_service.ts

```diff
@@ -379,16 +425,34 @@ export class SwapService {
 
     private async _getSwapQuotePartialTransactionAsync(
         swapQuote: SwapQuote,
-        isETHSell: boolean,
+        isFromETH: boolean,
+        isToETH: boolean,
         affiliateAddress: string,
+        swapVersion: SwapVersion,
     ): Promise<SwapQuoteResponsePartialTransaction> {
-        const extensionContractType = isETHSell ? ExtensionContractType.Forwarder : ExtensionContractType.None;
+        let extensionContractType;
+        switch (swapVersion) {
+            case SwapVersion.V0:
+                extensionContractType = isFromETH ? ExtensionContractType.Forwarder : ExtensionContractType.None;
+                break;
+            case SwapVersion.V1:
+                extensionContractType =
+                    isFromETH || isToETH ? ExtensionContractType.ExchangeProxy : ExtensionContractType.None;
```

Comment:
This should always be `ExtensionContractType.ExchangeProxy`.

The EP allows us to do more magic (withdraw WETH to ETH, deposit ETH to WETH), but it comes at a constant cost. We always use the EP in v1 (never the default). EP (FlashWallet) is always the taker.